src/include/migraphx/op/reduce_op.hpp

```diff
@@ -105,18 +106,39 @@ struct reduce_op : op_name<Derived>
         return tuned_axes;
     }
 
+    /**
+     * @brief returns a shape in which the axis or axes named
+     * for reduction by this op are set to size 1.
+     *
+     * @param inputs list of input shapes
+     * @return shape
+     */
     shape normalize_compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.has(1);
-        auto s          = inputs.at(0);
-        auto lens       = s.lens();
-        auto tuned_axes = tune_axes(lens.size());
-        for(auto axis : tuned_axes)
+        check_shapes{inputs, *this, true}.has(1);
+        auto s = inputs.at(0);
+        if(s.dynamic())
         {
-            lens[axis] = 1;
+            auto output_dyn_dims = s.dyn_dims();
+            auto tuned_axes      = tune_axes(output_dyn_dims.size());
```

Comment:
This can be moved above the `if(s.dynamic())` check. You can do `tune_axes(s.ndim())`, but I think you will need #1423.

Reply:
Not sure what you mean here...there is no ndim() in branch dyn_ref_multibroadcast either.  In any case, the idference would be insignificant.

Reply:
He meant that using `shape::ndims()` from my PR could simplify the code by moving the `tuned_axes` call outside of the if-else statement. Not a major issue either way.

Reply:
I couldn't find the definition of that function in the referenced branch, so let's just leave it as is for now.